modules/Logger.js

```diff
@@ -0,0 +1,169 @@
+(function(root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    define(['underscore', 'backbone', './ServiceCell'], factory);
+  } else if (typeof exports === 'object') {
+    module.exports = factory(require('underscore'), require('backbone'), require('./ServiceCell'));
+  } else {
+    root.Torso = root.Torso || {};
+    root.Torso.Logger = factory(root._, root.Backbone, root.Torso.ServiceCell);
+  }
+}(this, function(_, Backbone, ServiceCell) {
+  'use strict';
+
+  var Logger = ServiceCell.extend({ 
+
+    log : {}, 
+    sessions: [],
+    unclosed: {},
+    _isClosed: true,
+    _sessionStart: null,
+    _sessionEnd: null,
+    _sessionID: null,
+
+    /**
+    * saves information about event and wraps start and end signals in 'session' 
+    * @ param {JSON object} contains tracking information (including state: 'start' or 'end', uuid, sessionID)
+    * @ method track
+    */
+  	track: function(eventInfo){
+      if (eventInfo.state){
+        if (this._isClosed){
+          this.resetSession();
+          if (eventInfo.state == "start") {
+            this.openSession(eventInfo);
+            this.startEvent(eventInfo);
+          }
+        } else {
+          if (eventInfo.state == "start") {
+            this.startEvent(eventInfo);
+          }
+          else if (eventInfo.state == "end") {
+              this.endEvent(eventInfo);
+              if (Object.keys(this.unclosed).length === 0) {
+                this.closeSession();
+              }
+          }
+        }
+      }
+  	},
+
+    /**
+    * tracks information about the 'start' of events
+    * @param {JSON object} contains tracking information including UUID and time
+    * @method startEvent
+    */
+    startEvent: function(eventInfo){
+      var uuid = eventInfo.UUID;
+      eventInfo.sessionID = this._sessionID; //check if event info has sessionID property and throw error
+      eventInfo.startTime = eventInfo.time;
+      delete eventInfo.time;
+      this.unclosed[uuid] = eventInfo;
+    },
+
+    /**
+    * tracks information about the 'end' of events
+    * @param {JSON object} contains information about the tracked event, including UUID, time, state
+    * @method endEvent
+    */
+    endEvent: function(eventInfo){
+      var uuid = eventInfo.UUID;
+      var totalEvent = this.unclosed[uuid];
+      var duration = eventInfo.time - totalEvent.startTime; 
+      totalEvent.duration = duration;
+      totalEvent.endTime = eventInfo.time;
```

Comment:
Same with endTime. We probably can default the end time to when someone calls endEvent